### PR TITLE
Gmsh extension3

### PR DESCRIPTION
**Pull request by jotaraz**
Most important: Added functions for incomplete grids. (read from gmsh and seal)
Secondly: Added rudimentary functions for mixed grids

### CODE REVIEW

File: ext/ExtendableGridsGmshExt.jl
Comment (j-fu) [inline ext/ExtendableGridsGmshExt.jl:1]: Typo ?

**Comment by j-fu** [comment]
@cmerdon - could you try this PR ? IMHO it is ok.

**Comment by chmerdon** [comment]
good, it works for me. However, simplexgrid_from_gmsh seems to generate an ExtendalbeGrids{Float64, Int64} always, there seems to be no way to use Int32. Maybe we want to have that possibility?

**Comment by j-fu** [comment]
Hm @jotaraz, how does gmsh store integers ? Always as 64bit ?
May be we could add a kwarg `indextype` which by default would be what gmsh has.

**Comment by jotaraz** [comment]
@j-fu @chmerdon As far as I know, Gmsh (or at least Gmsh.jl) stores node indices etc. as UInt64 and the coordinates as Float64.

Since the last updates, there is the possibility to construct a grid from a gmsh file (or module) with any desired types for Ti, Tc. The conversion in the other direction (ExtendableGrid -> Gmsh) seems to work already, independently of the type of the ExtendableGrid.

**Comment by j-fu** [comment]
I would like to make 

```
ExtendableGrids.simplexgrid_from_gmsh
ExtendableGrids.simplexgrid_to_gmsh
ExtendableGrids.mixedgrid_from_gmsh
ExtendableGrids.mixedgrid_to_gmsh
```
public API for all the gmsh stuff. 
Maybe `mixed` and `simplex` could be merged.

As for the tests: Gmsh creates different grids on windows...

**Pull request merged**